<launch>

    <arg name="simulate" default="true"/>
    <arg name="record" default="true"/>

    <build_depend>message_generation</build_depend>
    <exec_depend>message_runtime</exec_depend>

    <group if="$(arg simulate)">
        <include file="$(find bluerov_sim)/launch/gazebo_base.launch"/>

        <include file="$(find bluerov_sim)/launch/spawn_vehicle.launch">
            <arg name="z" value="-0.3"/>
        </include>
    </group>

    <param name="simulate" value="$(arg simulate)" type="bool" />

    <node name="mixer" pkg="fav_depth_controller" type="mixer.py" output="screen" />

    <node name="experiment_parameters" pkg="fav_depth_controller" type="set_parameters.py" output="screen" />

    <node name="depth_setpoints" pkg="fav_depth_controller" type="depth_setpoint.py" output="screen" />
    
    <node name="state_estimator" pkg="fav_depth_controller" type="state_estimator.py" output="screen" />

    <node name="zController" pkg="fav_depth_controller" type="zController.py" output="screen" />

    <group if="$(arg record)">
        <node name="record" pkg="rosbag" type="record" args="-o $(find fav_depth_controller)/bags/bag /depth_setpoint /ground_truth/state /parameters /mavros/setpoint_motor/setpoint /state /vertical_thrust /control_error /pressure /rosout"/>
    </group>

</launch>
